### DISCUSSION TITLE
Publish application to F-Droid? (Official/izzysoft/self-hosted repo)

### DISCUSSION THREAD

**Discussion by hahalolha** [KoalaSat/nostros#40] [Ideas]
Excellent project, I myself am quite new to Nostr and having a native Android app is quite neat and useful. 

Although, unless I have missed it, it appears this app isn't listed on F-Droid.

[F-Droid](https://f-droid.org/en/about/) is an alternative Application Marketplace which is excellent for both its ease of use and discover-ability of new FOSS projects.

There are different methods for getting an application published:

1. Publish directly through F-Droid's official repo (info [here](https://f-droid.org/en/docs/Submitting_to_F-Droid_Quick_Start_Guide/)). F-Droid maintainers will compile the apk, verify its integrity  against releases and sign it with their own certificate. Updates may be slightly delayed but will be available to all users of F-Droid.

_Think of this as Debian's Stable Repo_

2. Publish to [IzzyOnDroid's Repo](https://android.izzysoft.de/repo/info) which accepts releases quicker due to a different ingest system (see §"How do apps come into this repo?" and §"What about updates?"). As this is a third-party repo, it doesn't come preinstalled bu default, meaning users must add this repo (although, this repo is quite popular, second only to the Official repo meaning most users likely already have this repo already configured).

_Think of this as Debian's Testing/Sid Repo_ 

3. Publish through your own [self-hosted repo](https://f-droid.org/en/docs/Setup_an_F-Droid_App_Repo). This method gives maintainers ultimate control of versioning, releases, signing and provides the quickest time from release to update. This method is probably not ideal as it would require users to discover this app elsewhere and add it to their client. 

_Think of this as a PPA Repo_

Ultimately, all these options achieve the same thing, simplifying the install (and discovery) process. I believe this is a worthwhile investment of time. 

_Please note, most of this post was written from the top of my head and may contain errors. Please utilise referenced documentation for further details._

**Comment by KoalaSat** [comment]
Thank you for so much info @hahalolha, publishing this app on F-Droid is certainly on the roadmap, but I consider it's way too early for an official release (we are still on alpha and there are certain basic functionalities plus deep testing to be done). Probably, this will happen once we move to beta. 

Anyways, I'll keep this to use it as source of info once we are ready, thanks again!
